pkg/parser/parser.go

```diff
@@ -321,28 +315,28 @@ func getCerts(log logrus.FieldLogger, s store.Storer, secretsToSNIs map[string][
 
 func getServiceEndpoints(log logrus.FieldLogger, s store.Storer, svc corev1.Service,
 	servicePort *corev1.ServicePort) []kongstate.Target {
-	var targets []kongstate.Target
-	var endpoints []util.Endpoint
 
 	log = log.WithFields(logrus.Fields{
 		"service_name":      svc.Name,
 		"service_namespace": svc.Namespace,
 		"service_port":      servicePort,
 	})
 
-	endpoints = getEndpoints(log, &svc, servicePort, corev1.ProtocolTCP, s.GetEndpointsForService)
+	// in theory a Service could have multiple port protocols, we need to ensure we gather
+	// endpoints based on all the protocols the service is configured for. We always check
+	// for TCP as this is the default protocol for service ports.
+	protocols := listProtocols(svc)
+
+	// check all protocols for associated endpoints
+	endpoints := []util.Endpoint{}
```

Comment:
```suggestion
	var endpoints []util.Endpoint
```
https://gist.github.com/adamveld12/c0d9f0d5f0e1fba1e551#declaring-empty-slices